lib/cn/node_split.c

```diff
@@ -193,23 +203,39 @@ find_inflection_key(
          * a max key >= the min key of the inflection kblock. At this kblock,
          * a forward WBT leaf node iterator needs to seek toward the inflection
          * kblock's min key.
-         *
-         * In the case no kblock's max key is >= the inflection kblock's min
-         * key, then we can skip the forward WBT leaf node iteration in the next
-         * step.
          */
-        while (++iter->offset < iter->ks->ks_st.kst_kblks) {
-            curr += 1;
-
-            res = keycmp(curr->kb_koff_max, curr->kb_klen_max, *inflection_key,
-                *inflection_key_len);
+        for (; iter->offset < iter->ks->ks_st.kst_kblks; iter->offset++, curr++) {
```

Comment:
nit: Maybe change the comment to somthing like this:

> If the current kblock lies entirely to the left of the inflection key, go to the next kblock since the split key must be greater than or equal to the inflection key.

This way it's more about why you're doing this vs explaining what you're doing.

Another nit: I don't believe this needs to be a for loop. The curr+1 kblock will, by definition, have a min key >= the inflection key. If that wasn't the case, this kvset's iterator would have been positioned at curr+1 and not at curr. So the max key of curr+1 will also be >= the inflection key.

You may be able to avoid doing 2 keycmp where one is enough:

```
if (iter->offset < iter->ks->ks_st.kst_kblks - 1 &&
     keycmp(curr->kb_koff_max, curr->kb_klen_max, inflection_kblk->kb_koff_min, inflection_kblk->kb_klen_min) <= 0) {
    iter->offset++;
    curr++;
}
```